plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectLineRecordReader.java

```diff
@@ -110,7 +120,47 @@ public S3SelectLineRecordReader(
         closer.register(selectClient);
     }
 
-    public abstract SelectObjectContentRequest buildSelectObjectRequest(Properties schema, String query, Path path);
+    public abstract InputSerialization buildInputSerialization();
+
+    public abstract OutputSerialization buildOutputSerialization();
+
+    public abstract boolean shouldEnableScanRange();
+
+    protected Properties getSchema()
+    {
+        return schema;
+    }
+
+    protected CompressionType getCompressionType()
+    {
+        return compressionType;
+    }
+
+    public SelectObjectContentRequest buildSelectObjectRequest(String query, Path path)
+    {
+        SelectObjectContentRequest selectObjectRequest = new SelectObjectContentRequest();
+        URI uri = path.toUri();
+        selectObjectRequest.setBucketName(TrinoS3FileSystem.extractBucketName(uri));
+        selectObjectRequest.setKey(TrinoS3FileSystem.keyFromPath(path));
+        selectObjectRequest.setExpression(query);
+        selectObjectRequest.setExpressionType(SQL);
+
+        InputSerialization selectObjectInputSerialization = buildInputSerialization();
+        selectObjectRequest.setInputSerialization(selectObjectInputSerialization);
+
+        OutputSerialization selectObjectOutputSerialization = buildOutputSerialization();
+        selectObjectRequest.setOutputSerialization(selectObjectOutputSerialization);
+
+        if (shouldEnableScanRange()) {
+            // Scan range queries are not supported for compressed files.
```

Comment:
Why do we need to document here this limitation?

Reply:
I think that the `shouldEnableScanRange()` check explains the reasoning behind this choice pretty well.

Reply:
Updated